be/test/fs/fs_s3_test.cpp

```diff
@@ -310,4 +311,44 @@ TEST_F(S3FileSystemTest, test_delete_nonexist_file) {
     ASSERT_OK(fs->delete_file(S3Path("/nonexist.dat")));
 }
 
+TEST_F(S3FileSystemTest, test_new_S3_client_with_rename_operation) {
+    ASSIGN_OR_ABORT(auto fs, FileSystem::CreateUniqueFromString("s3://"));
+    // only used for generate a new S3 client into global cache
+    (void)fs->rename_file(S3Path("/dir/source_name"), S3Path("/dir/target_name"));
+
+    // basic config
+    Aws::Client::ClientConfiguration config = S3ClientFactory::getClientConfig();
+    S3URI src_uri;
+    ASSERT_TRUE(src_uri.parse(S3Path("/dir/source_name")));
+    if (!src_uri.endpoint().empty()) {
+        config.endpointOverride = src_uri.endpoint();
+    } else if (!config::object_storage_endpoint.empty()) {
+        config.endpointOverride = config::object_storage_endpoint;
+    } else if (config::object_storage_endpoint_use_https) {
+        config.scheme = Aws::Http::Scheme::HTTPS;
+    } else {
+        config.scheme = Aws::Http::Scheme::HTTP;
+    }
+    if (!config::object_storage_region.empty()) {
+        config.region = config::object_storage_region;
+    }
+    config.maxConnections = config::object_storage_max_connection;
+    if (config::object_storage_connect_timeout_ms > 0) {
+        config.connectTimeoutMs = config::object_storage_connect_timeout_ms;
+    }
+
+    // reset requestTimeoutMs as config::object_storage_rename_file_request_timeout_ms
+    // to check hit the cache or not.
+    config.requestTimeoutMs = config::object_storage_rename_file_request_timeout_ms;
+    ASSERT_TRUE(S3ClientFactory::instance().find_client_cache_keys_by_config_TEST(config));
+
+    // use config::object_storage_request_timeout_ms instead
+    config::object_storage_rename_file_request_timeout_ms = -1;
+    config::object_storage_request_timeout_ms = 1000;
```

Comment:
remember to reset the config var to default value, don't affect other test cases in case their correctness depends on the default config value.

Reply:
Updated